### PR TITLE
fix: changed search_node action to determine correct index

### PR DESCRIPTION
**Pull request by be-ndee**
see #1014

### CODE REVIEW

File: lua/nvim-tree/actions/search-node.lua [hunk 1]
```diff
@@ -7,6 +7,7 @@ local M = {}
 function M.fn()
   if not TreeExplorer then return end
 
+  -- ask user for path
```
Comment (kyazdani42) [inline lua/nvim-tree/actions/search-node.lua:10]: please remove these comments, we can read the code well :smile:

File: lua/nvim-tree/actions/search-node.lua [hunk 2]
```diff
@@ -15,49 +16,74 @@ function M.fn()
...
-    local index_partial_match = 0
-    for _, node in ipairs(nodes) do
-      index_partial_match = index_partial_match + 1
 
+      -- if node is directory -> go through directory
       if node.nodes then
-        local matches = utils.str_find(absolute_input_path, node.absolute_path)
+        -- check if node matches with path
+        -- append "/" to name, so files do not match with directory-paths
+        -- e.g. user searches for "/foo/bar.txt", than directory "/foo/bar" should not match with filename
```
Comment (kyazdani42) [inline lua/nvim-tree/actions/search-node.lua:53]: You can keep only the 2nd line of the comment :)

File: lua/nvim-tree/actions/search-node.lua [hunk 3]
```diff
@@ -15,49 +16,74 @@ function M.fn()
...
         end
       end
+
+      if node.open then
+        index = index + count_visible_nodes(node.nodes)
+      end
     end
 
-    return 0
+    return index;
```
Comment (kyazdani42) [inline lua/nvim-tree/actions/search-node.lua:75]: leading `;` not needed

**Review by kyazdani42 — CHANGES_REQUESTED** [review]
looks good to me, except those comments are not needed :)

**Comment by be-ndee** [comment]
I've done the requested changes :)

**Review by kyazdani42 — APPROVED** [review]

**Pull request merged**